src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEParameterSymbol.vb

```diff
@@ -605,6 +608,30 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             End Get
         End Property
 
+        Friend Overrides ReadOnly Property CallerArgumentExpressionParameterIndex As Integer
+            Get
+                If _lazyCallerArgumentExpressionParameterIndex = -2 Then
+                    Debug.Assert(Not _handle.IsNil)
+
+                    Dim attribute = PEModule.FindTargetAttribute(_handle, AttributeDescription.CallerArgumentExpressionAttribute)
+                    Dim parameterName As String = Nothing
+                    If attribute.HasValue AndAlso PEModule.TryExtractStringValueFromAttribute(attribute.Handle, parameterName) Then
+                        Dim parameters = ContainingSymbol.GetParameters()
+                        For i = 0 To parameters.Length - 1
+                            If IdentifierComparison.Equals(parameters(i).Name, parameterName) Then
```

Comment:
>IdentifierComparison.Equals(parameters(i).Name, parameterName) [](http://example.com/codeflow?start=31&length=62)

Couple of notes:

1. I think the C# version of this method is broken, as it's using default string comparison. It should be using `Ordinal`.
2. This is potentially going to mean differing behavior between C# and VB. I think we likely should be using Ordinal here as well, not case-insensitive comparison. If VB defined a parameter named `param` but referred to it as `Param` in the attribute, we would accept that here, but not in C#.